main.js

```diff
@@ -1,2 +1,122 @@
+    'History': document.querySelector('.calculator-history')
+  }
+
+  this.handleNumberInput = function(numberInput) {
+    var nodeToAdd
+    if( this.domElements.CurrentInput.children.length > 0 ) { // If we already have an ongoing input
+      nodeToAdd = this.domElements.CurrentInput.firstChild
+      nodeToAdd.textContent += numberInput
+    } else {
+      nodeToAdd = document.createElement('p')
+      nodeToAdd.textContent = numberInput
+    }
+    nodeToAdd.style.margin = 0
+    if( this.domElements.History.children.length > 4 ) {
+      this.domElements.History.removeChild( this.domElements.History.firstChild )
+    }
+    if( this.domElements.CurrentInput.children.length > 0 ) {
+      this.domElements.History.appendChild( this.domElements.CurrentInput.firstChild )
+    }
+    this.domElements.CurrentInput.appendChild( nodeToAdd )
+  }
+
+  this.handleOperatorInput = function(operatorInput) {
+    console.log(operatorInput)
+  }
+
+  this.keyDownInterpreter = function(keyEvent) {
+    // Prevent double click when pressing enter
+    var activeElement = document.activeElement
+    activeElement.blur()
+
+    var getOperatorFromCode = {
+      190: 'Decimal',   // .
+      88:  'Multiply',  // x
+      191: 'Divide',    // /
+      189: 'Minus',     // -
+      187: 'Equals',    // =
+      13:  'Equals',    // enter
+      27:  'Clear',     // esc
+      67:  'Clear',     // c
+      70:  'Flip'       // f
+    }
+    var keyCode = keyEvent.keyCode;
+
+    if( keyCode >= 48 && keyCode <= 57 ) { // 48 is 0, 57 is 9. between is other numbers
+      if( keyEvent.shiftKey && keyCode === 56 ) {
+        this.handleOperatorInput.call(this, 'Multiply')
+      } else if( keyEvent.shiftKey && keyCode === 53 ) {
+        this.handleOperatorInput.call(this, 'Percentage')
+      } else {
+        this.handleNumberInput.call(this, keyCode-48)
+      }
+    } else {
+      if( keyEvent.shiftKey && keyCode === 187 ) {
+        this.handleOperatorInput.call(this, 'Plus')
+      } else if( getOperatorFromCode.hasOwnProperty(keyCode) ) {
+        this.handleOperatorInput.call(this, getOperatorFromCode[keyCode] + keyEvent)
+      } else {
+        // NOTE: Remove these lines before posting to prrr; for debug only
+        // console.log('Keycode: ', keyCode)
+      }
+    }
+
+    // Give focus back to element now that key has been handeled
+    // timeout required so that enter doesn't continue to activate focused button
+    window.setTimeout( function() {
+      activeElement.focus()
+    }, 1 ) // Even a single millisecond is enough to consistantly stop the problem. Amazing.
+  }
+
+  this.initialize = function() {
+    // Handle keyboard events
+    document.addEventListener('keydown', this.keyDownInterpreter.bind(this), false)
+
+    // Populate onclick handlers for calculator buttons
+    for(var i = 0; i < 10; i++) {
+      this.domElements.Numbers[ i ].onclick = Calculator.scopeLock.call(this, i, this.handleNumberInput.bind(this) )
```

Comment:
Not sure what spacing convention to use. If I close the gap, then I can't Cmd D to highlight all instances of 'i' because 'i' is a single letter and appears in the middle of keywords and other variables. Should I just change the name from 'i' to 'iteratorI' ? What do other people do to deal with this problem?